framework/src/base/EigenProblem.C

```diff
@@ -162,3 +162,12 @@ EigenProblem::converged()
 {
   return _nl_eigen->converged();
 }
+
+bool
+EigenProblem::isNonlinearEigenvalueSolver()
+{
+  return solverParams()._eigen_solve_type == Moose::EST_NONLINEAR_POWER ||
+         solverParams()._eigen_solve_type == Moose::EST_MF_NONLINEAR_POWER ||
+         solverParams()._eigen_solve_type == Moose::EST_MONOLITH_NEWTON ||
+         solverParams()._eigen_solve_type == Moose::EST_MF_MONOLITH_NEWTON;
```

Comment:
Notice that you are not using `_is_nonlinear_eigenvalue_solver` at all. If you know `solverParams()._eigen_solve_type` during construction, then I say set `_is_nonlinear_eigenvalue_solver` there and have this method return its value.

If you have to wait for `solverParams()._eigen_solve_type` to be set, then get rid of `_is_nonlinear_eigenvalue_solver`.